homeassistant/components/vicare/button.py

```diff
@@ -45,28 +45,41 @@ class ViCareButtonEntityDescription(
 )
 
 
+def _build_entity(name, vicare_api, device_config, description):
+    """Create a ViCare button entity."""
+    _LOGGER.debug("Found device %s", name)
+    try:
+        description.value_getter(vicare_api)
+        _LOGGER.debug("Found entity %s", name)
+    except PyViCareNotSupportedFeatureError:
+        _LOGGER.info("Feature not supported %s", name)
+        return None
+    except AttributeError:
+        _LOGGER.debug("Attribute Error %s", name)
+        return None
+
+    return ViCareButton(
+        name,
+        vicare_api,
+        device_config,
+        description,
+    )
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Create the ViCare binary sensor devices."""
+    """Create the ViCare button entities."""
     name = VICARE_NAME
     api = hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
 
     entities = []
 
     for description in BUTTON_DESCRIPTIONS:
```

Comment:
It's better to create all entities once in the executor so we don't switch context multiple times. Switching context is expensive.

Reply:
Agreed. Can we do this in a separate PR again?
First of all there is only a single button in this integration. Second: We have this pattern of creating all entities in individual executor calls so it would end up in a larger refactoring. I'm happy to do that but it might be out of scope for this PR

Reply:
Ok.